navigation-material/src/androidTest/java/com/google/accompanist/navigation.material/HalfExpandedTests.kt

```diff
@@ -0,0 +1,102 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.navigation.material
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.ModalBottomSheetValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.plusAssign
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import com.google.common.truth.Truth
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+@OptIn(ExperimentalMaterialApi::class, ExperimentalMaterialNavigationApi::class)
+class HalfExpandedTests {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun testSkipHalfExpandedWhenOverHalfScreenHeight(): Unit = runBlocking {
+        trySkipHalfExpanded(true, 0.6f, ModalBottomSheetValue.Expanded)
+    }
+
+    @Test
+    fun testNoSkipHalfExpandedWhenOverHalfScreenHeight(): Unit = runBlocking {
+        trySkipHalfExpanded(false, 0.6f, ModalBottomSheetValue.HalfExpanded)
+    }
+
+    @Test
+    fun testNoSkipHalfExpandedWhenUnderHalfScreenHeight(): Unit = runBlocking {
+        trySkipHalfExpanded(false, 0.4f, ModalBottomSheetValue.Expanded)
+    }
+
+    @Test
+    fun testSkipHalfExpandedWhenUnderHalfScreenHeight(): Unit = runBlocking {
+        trySkipHalfExpanded(true, 0.4f, ModalBottomSheetValue.Expanded)
+    }
+
+    private suspend fun trySkipHalfExpanded(skipHalfExpanded: Boolean, screenPercent: Float, expectedSheetState: ModalBottomSheetValue) {
+        coroutineScope {
+            lateinit var navController: NavHostController
+            lateinit var bottomSheetNavigator: BottomSheetNavigator
+            composeTestRule.setContent {
+                navController = rememberNavController()
+                bottomSheetNavigator = rememberBottomSheetNavigator(skipHalfExpanded = skipHalfExpanded)
+
+                navController.navigatorProvider += bottomSheetNavigator
+
+                NavHost(
+                    navController = navController,
+                    startDestination = "start"
+                ) {
+                    composable("start") {
+                        ModalBottomSheetLayout(
```

Comment:
I hadn't considered this usage of the API before! While it works for this test because the `bottomSheet` destination is a `FloatingWindow` that is displayed above the `start` destination, this would break if we navigated to another non-floating destination because the `ModalBottomSheetLayout` would leave the composition.

Is there a reason we don't wrap the `NavHost` with the `ModalBottomSheetLayout` (see the [docs](https://google.github.io/accompanist/navigation-material/#bottom-sheet-destinations))?